src/lzfse_decode.c

```diff
@@ -19,16 +19,18 @@ HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABI
 ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 
+#include <stdio.h>
+
 // LZFSE decode API
 
 #include "lzfse.h"
 #include "lzfse_internal.h"
 
 size_t lzfse_decode_scratch_size() { return sizeof(lzfse_decoder_state); }
 
-size_t lzfse_decode_buffer(uint8_t *__restrict dst_buffer, size_t dst_size,
-                           const uint8_t *__restrict src_buffer,
-                           size_t src_size, void *__restrict scratch_buffer) {
+size_t lzfse_do_decode_buffer(uint8_t *__restrict dst_buffer, size_t dst_size,
```

Comment:
I don't like the 'do_decode'. What about lzfse_decode_buffer_<something>. 'with_scratch', 'no_alloc', or any suggestions? Same for do_encode.